src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECDsaCng.ImportExport.cs

```diff
@@ -0,0 +1,558 @@
+            //     byte[cbCofactor]             H
+            //     byte[cbSeed]                 S
+            //     byte[cbFieldLength]          Qx
+            //     byte[cbFieldLength]          Qy
+            //     -- Private only --
+            //     byte[cbSubgroupOrder]        D
+            byte[] ecBlob = ExportFullKeyBlob(key, includePrivateParameters);
+            ECParameters ecParams = new ECParameters();
+
+            KeyBlobMagicNumber magic = (KeyBlobMagicNumber)BitConverter.ToInt32(ecBlob, 0);
+
+            // Check the magic value in the key blob header. If the blob does not have the required magic,
+            // then throw a CryptographicException.
+            CheckMagicValueOfKey(magic, includePrivateParameters);
+
+            var primeCurve = new ECPrimeCurve();
+            ecParams.Curve = primeCurve;
+
+            unsafe
+            {
+                // Fail-fast if a rogue provider gave us a blob that isn't even the size of the blob header.
+                if (ecBlob.Length < sizeof(BCRYPT_ECCFULLKEY_BLOB))
+                    throw ErrorCode.E_FAIL.ToCryptographicException();
+
+                fixed (byte* pEcBlob = ecBlob)
+                {
+                    BCRYPT_ECCFULLKEY_BLOB* pBcryptBlob = (BCRYPT_ECCFULLKEY_BLOB*)pEcBlob;
+
+                    primeCurve.CurveType = (ECCurveType)pBcryptBlob->CurveType;
+
+                    int offset = sizeof(BCRYPT_ECCFULLKEY_BLOB);
+
+                    primeCurve.P = Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+                    primeCurve.A = Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+                    primeCurve.B = Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+                    primeCurve.G.X = Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+                    primeCurve.G.Y = Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+                    primeCurve.N = Consume(ecBlob, ref offset, pBcryptBlob->cbSubgroupOrder);
+                    primeCurve.H = Consume(ecBlob, ref offset, pBcryptBlob->cbCofactor);
+                    primeCurve.S = Consume(ecBlob, ref offset, pBcryptBlob->cbSeed);
+
+                    ECPoint qpoint = new ECPoint();
+                    ecParams.Q = qpoint;
+                    qpoint.X = Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+                    qpoint.Y = Consume(ecBlob, ref offset, pBcryptBlob->cbFieldLength);
+
+                    if (includePrivateParameters)
+                    {
+                        ecParams.D = Consume(ecBlob, ref offset, pBcryptBlob->cbSubgroupOrder);
+                    }
+                }
+            }
+
+            return ecParams;
+        }
+
+        #endregion
+
+        /// <summary>
+        ///     Append "value" to the data already in blob.
+        /// </summary>
+        private static void Emit(byte[] blob, ref int offset, byte[] value)
```

Comment:
I'm pretty sure these are copied from RSA... Can we put them somewhere sharable?